src/webots/maths/WbMathsUtilities.hpp

```diff
@@ -63,4 +67,25 @@ inline double WbMathsUtilities::normalizeAngle(double angle, double lastSpot = 0
     d -= 2.0 * M_PI;
   return d + lastSpot;
 }
+
+// Make sure that the value is within the valid range before calling acos
+// If not then the behavior is similar to a clamped value.
+inline double WbMathsUtilities::clampedAcos(double value) {
+  if (value >= 1.0)
+    return 0.0;
+  else if (value <= -1.0)
```

Comment:
The `else` is not needed here.
```suggestion
  if (value <= -1.0)
```